lib/drivers/medtronic/processData.js

```diff
@@ -272,24 +272,39 @@ var buildWizardRecords = function (records) {
     bgRaw = bgRaw.concat(struct.extractByte(record.head,1));
     bgInput = struct.extractBEShort(bgRaw,0);
 
-    var carbRaw = [(struct.extractByte(record.body,1) & BITMASKS.RIGHT_1100) >> 2];
+    var carbRaw = [struct.extractByte(record.body,1) >> 2 & 1];
     carbRaw = carbRaw.concat(struct.extractByte(record.body,0));
     carbInput = struct.extractBEShort(carbRaw,0);
 
-    var carbRatioRaw = [struct.extractByte(record.body,2) & BITMASKS.RIGHT_THREE];
+    var carbRatioRaw = [0,struct.extractByte(record.body,9) >> 1 & BITMASKS.RIGHT_TWO];
+    carbRatioRaw = carbRatioRaw.concat(struct.extractByte(record.body,2));
     carbRatioRaw = carbRatioRaw.concat(struct.extractByte(record.body,3));
-    carbRatio = struct.extractBEShort(carbRatioRaw,0) / 10.0;
+    carbRatio = struct.extractBEInt(carbRatioRaw,0) / 10.0;
 
     bgTarget.low = struct.extractByte(record.body,5);
-    isf = struct.extractByte(record.body,4);
     bgTarget.high = struct.extractByte(record.body,14);
-    recommended.carb = struct.extractBEShort(record.body,7) / settings.strokesPerUnit;
 
-    recommended.correction = twosComplement((((struct.extractByte(record.body,9) & BITMASKS.LEFT_00111) << 5) + struct.extractByte(record.body,6)),11) / settings.strokesPerUnit;
+    var isfRaw = [struct.extractByte(record.date,2) >> 6 & BITMASKS.RIGHT_TWO];
+    isfRaw = isfRaw.concat(struct.extractByte(record.body,4));
+    isf = struct.extractBEShort(isfRaw,0);
 
-    recommended.net = struct.extractBEShort(record.body,12) / settings.strokesPerUnit;
+    var rawRecommendedCarb = [0,struct.extractByte(record.body,9) & 1];
+    rawRecommendedCarb = rawRecommendedCarb.concat(struct.extractByte(record.body,7));
+    rawRecommendedCarb = rawRecommendedCarb.concat(struct.extractByte(record.body,8));
+    recommended.carb = struct.extractBEInt(rawRecommendedCarb,0) / settings.strokesPerUnit;
```

Comment:
Huh, so this is actually a 4-byte integer, where I previously had it as a 2-byte short.